In [4]:
install.packages("wooldridge")
library(wooldridge)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [0]:
attach(wage2)

In [6]:
lm1<-lm(log(wage)~educ+exper+tenure)
summary(lm1)
cov(exper, tenure)


Call:
lm(formula = log(wage) ~ educ + exper + tenure)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.8282 -0.2401  0.0203  0.2569  1.3400 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 5.496696   0.110528  49.731  < 2e-16 ***
educ        0.074864   0.006512  11.495  < 2e-16 ***
exper       0.015328   0.003370   4.549 6.10e-06 ***
tenure      0.013375   0.002587   5.170 2.87e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3877 on 931 degrees of freedom
Multiple R-squared:  0.1551,	Adjusted R-squared:  0.1524 
F-statistic: 56.97 on 3 and 931 DF,  p-value: < 2.2e-16


[1] 5.409597

#Linear Regression
#####**_$log(wage)$_**= 5.4967 + 0.0749 **_$educ$_** + 0.0153 **_$exper$_** + 0.0134 **_$tenure$_**

#####**_$R^{2}$_** = 0.15
#####**_$N$_**= 935

####**_$\beta$<sub>2</sub>_** ---> Standard error= 0.0034
####**_$\beta$<sub>3</sub>_** ---> Standard error= 0.0026

In [7]:
lm2<-lm(log(wage)~educ)
summary(lm2)


Call:
lm(formula = log(wage) ~ educ)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.94620 -0.24832  0.03507  0.27440  1.28106 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 5.973062   0.081374   73.40   <2e-16 ***
educ        0.059839   0.005963   10.04   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.4003 on 933 degrees of freedom
Multiple R-squared:  0.09742,	Adjusted R-squared:  0.09645 
F-statistic: 100.7 on 1 and 933 DF,  p-value: < 2.2e-16


In [8]:
anova(lm1, lm2)

Res.Df RSS      Df Sum of Sq F        Pr(>F)      
1 931    139.9610 NA        NA       NA           NA
2 933    149.5186 -2 -9.557619 31.78795 4.422016e-14

In [9]:
ssr.lm1 = sum((summary(lm1)$residuals)^2)
ssr.lm2 = sum((summary(lm2)$residuals)^2)
ssr.lm1
ssr.lm2

[1] 139.961

[1] 149.5186

In [10]:
## P-value ##
pf(31.7880, 2, 931, lower.tail=FALSE)

[1] 4.4218e-14

In [11]:
## Critical Value##
qf(0.95, 2, 931)

[1] 3.005393

#####Anova shows that the model lm1 is is significantly different from model lm2. That means that $\beta$<sub>2</sub> and $\beta$<sub>3</sub> are jointly significant.

In [12]:
install.packages("lmtest")
library(lmtest)
install.packages("plm")
library(plm)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependency ‘zoo’


Loading required package: zoo


Attaching package: ‘zoo’


The following objects are masked from ‘package:base’:

    as.Date, as.Date.numeric


Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

also installing the dependencies ‘miscTools’, ‘bibtex’, ‘gbRd’, ‘bdsmatrix’, ‘sandwich’, ‘maxLik’, ‘Rdpack’, ‘Formula’




In [16]:
coeftest(lm1)
coeftest(lm1, vcov=vcovHC, type="HC0") 


t test of coefficients:

             Estimate Std. Error t value  Pr(>|t|)    
(Intercept) 5.4966957  0.1105282 49.7311 < 2.2e-16 ***
educ        0.0748638  0.0065124 11.4955 < 2.2e-16 ***
exper       0.0153285  0.0033696  4.5491 6.100e-06 ***
tenure      0.0133748  0.0025872  5.1696 2.871e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1



t test of coefficients:

             Estimate Std. Error t value  Pr(>|t|)    
(Intercept) 5.4966957  0.1117899 49.1699 < 2.2e-16 ***
educ        0.0748638  0.0066394 11.2757 < 2.2e-16 ***
exper       0.0153285  0.0033980  4.5110 7.275e-06 ***
tenure      0.0133748  0.0026517  5.0438 5.485e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


#####t($\beta$<sub>1</sub>)= 11.49. It is statistically significant at $p$ < 0.05
#####The value of p is 2.2e-16


In [20]:
expten=exper+tenure
expten

[1] 13 27 20 20 19 16 13 22 14 32 21 20  7  9 18 19 15 29 11 20 10 14 17 25 19
 [26] 17 21 12 13 21 19 19 14 20 17 19 12 23 11 18 32 19 20 27 24 24 18 16 21 26
 [51] 36 20 22 19 12 20 12 19  8 29 17 24 31 19 30 20 12 13 30 10 21 19 21 12 15
 [76] 14 23 32 10 15 19 22 18 17 18 14 29  6  9 16 21 13 19 17 16 25 21 21 12 17
[101] 18 18 32 18 17 31 15 17 12 15 16 33 13 21 21 18 21 21 18 30 23 38 26 14 20
[126] 13 21 18 25 21 13 11 14  6 17 21 18 23 31 16 31 16 21 13 33 29 13 15 19 19
[151] 22  7 14 13 12 31  8 23 19 10 19 22 23 12 33 17  8 27 31 11 17 25 18 17 16
[176] 15 14 21 17 17 17 20  8 19 20 18  8 21 25 30 17 34 11  2 18 16 20 27 18 21
[201] 16 11 25 13 23 29 33 15 15 14 10 18  7  8 18 18 22 22 19 20 35 23 36 17 22
[226] 16 17 17 19 20  8 16 27 22 10 16 11  9 24 19 12 13 15 25  9 21 23 24 34 26
[251] 11 28 31 17 26 16 31 14 22 15 11 17  7 13 11 22 13 35 10  8 12 20 13 21 16
[276] 11 10 24 15 19 15  6 28 28 25 16 18 20 11 28  9 27 40 27 22 33 15 29 35 25
[301] 27 25 18 10 30 32 10 25 14 17 12  8 11 27 15 19 29 42 31 23 14 18 11 25 16
[326] 13 14 13 27 14 25 15 31 20 33  6 23 22 14 15 13 28 19 13 13 10 37 18 31 20
[351] 20 18 34 13 10 25 23 28 16 11 26 10 19 14 22  9 33 12 11 35 21 22 18 15  8
[376] 10 21 10 26 20 18 16 23 15 16 13  8 30 35 27 27 23 11 11 33  9 35 12 17 39
[401] 16 16 13 30 17 18 14 24 11 12 20 23 22 27 13 14 15  9  8 21 29 15 18 34 24
[426]  5 17 24 22 18 15 14 24 11 22 14 19 11 27 34 31 19 20 14 20 16  9 21 10 18
[451] 33 10 15 35 17 33 13 16 14 17 13 19 13 29 30 21 22 20 35 21 11 14 23 26 31
[476] 19 22 16 10 20 19 32 21 22 31 12 14 14 35 12 30 14 28 17 30 20 29 22 25 37
[501] 16 14 11 19 20 19 13 35 22 17 20 22 20 26 13 18 21 17 15 18 12  8 18 25 25
[526] 19 11  8 20 17 31 24 12 11 33  9 15 28 18 18  8 21  8  9 17 13 10 24 27 26
[551] 15 26  8 27 17 22 30 25 15 14  4 24 16 28 21 14 20 11 30 29 13 12 19 17 20
[576] 30  9 28 26 16 33 10 21 12 16 14 22 34 10 21  8 36  5  8 15 12 25 15 23 20
[601] 11 12 24 14 15 35 32 23 26 10 35 11 15 16 12 22  7 22  9 13 21 10 17 25 12
[626] 22 13  8 20 16 18 17 15 26 17 16 15 21 21 10 27 20 11 26 16 23 17  9 12 17
[651] 22 17 16 29 22 18  6 26 19 10 25 15 15 17 17 13 19 31  8 27 30 13 18 25 24
[676] 23 15 21 34 10 31 11 33 19 17 13 21 17 13 26 30 15 15 30 30 15  7 10 11 14
[701] 13  6 10 17 15 23  2 22 29 24 39 19 21 18 27 18 24 10 15 16  5  6 16  7 16
[726] 21 12 23 17 19 19  9 13 13 12  5 14 30 23 12 14 14 18 22 23 13 13 14  7 11
[751] 11 20 11 17  2  8 29 14 17  7 13 10 25  8 18 17 28 24 34 16 24 22 23 31 19
[776] 21 16 19  8 11 18 10 16 12 10 27 27 32 12 27 32 21 28 13 13 27 24 10 20 12
[801] 18 18 36 16 18 12 12 12 26 20 28 25 23 22 22 30 18 10 14 21 16 15 13 16 10
[826] 16  6 23 39 29  2 20  6 15 21 11 15  8 17 15 30 13 14 23 33 21 27 14 21 27
[851] 24  8  5 25 20 25 20 16 34 13  7  8 18  8 29 39 21  9  5 16  7 23 34 13 19
[876] 27 26 13 29 28 14  9 22 12 16 29 18 17 12 19 16 21 14 33 19 26 19 28 21 14
[901] 15 23 22 20 12 27 32 23 12 16 33 30  5  9 14  9 14  7 17  9  9 27  7 11 18
[926] 14 21 16 12 19  7 13 22 24 35

In [23]:
lm3<-lm(log(wage)~educ+exper+expten)
summary(lm3)


Call:
lm(formula = log(wage) ~ educ + exper + expten)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.8282 -0.2401  0.0203  0.2569  1.3400 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)    
(Intercept) 5.496696   0.110528  49.731  < 2e-16 ***
educ        0.074864   0.006512  11.495  < 2e-16 ***
exper       0.001954   0.004743   0.412    0.681    
expten      0.013375   0.002587   5.170 2.87e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.3877 on 931 degrees of freedom
Multiple R-squared:  0.1551,	Adjusted R-squared:  0.1524 
F-statistic: 56.97 on 3 and 931 DF,  p-value: < 2.2e-16


Not Statistically significant at 5%